metricbeat/mb/builders.go

```diff
@@ -81,6 +95,10 @@ func NewModules(config []*common.Config, r *Register) (map[Module][]MetricSet, e
 		return nil, errs.Err()
 	}
 
+	if len(modToMetricSets) == 0 {
+		return nil, ErrAllModulesDisabled
```

Comment:
That is fine for now but I think we probably have to remove it in the future. The reasoning here is that in case centralised configuration is used, it could be that metricbeat is started without any modules enabled, and then later modules are enabled from the central config.